exp/runtime/internal/controllers/extensionconfig_controller_test.go

```diff
@@ -308,3 +310,101 @@ func fakeExtensionServer(discoveryHandler func(w http.ResponseWriter, r *http.Re
 	srv := httptest.NewServer(mux)
 	return srv
 }
+
+func Test_reconcileCABundle(t *testing.T) {
+	g := NewWithT(t)
+
+	scheme := runtime.NewScheme()
+	g.Expect(corev1.AddToScheme(scheme)).To(Succeed())
+
+	tests := []struct {
+		name         string
+		client       client.Client
+		config       *runtimev1.ExtensionConfig
+		wantCABundle []byte
+		wantErr      bool
+	}{
+		{
+			name:    "No-op because no annotation is set",
+			client:  fake.NewClientBuilder().WithScheme(scheme).Build(),
+			config:  fakeCAInjectionRuntimeExtensionConfig("some-extension-config", "some-namespace", "", ""),
+			wantErr: false,
+		},
+		{
+			name: "Inject ca-bundle",
+			client: fake.NewClientBuilder().WithScheme(scheme).WithObjects(
+				fakeCASecret("some-ca-secret", "some-namespace", "some-ca-data"),
+			).Build(),
+			config:       fakeCAInjectionRuntimeExtensionConfig("some-extension-config", "some-namespace", "some-namespace/some-ca-secret", ""),
+			wantCABundle: []byte(`some-ca-data`),
+			wantErr:      false,
+		},
+		{
+			name: "Inject ca-bundle",
+			client: fake.NewClientBuilder().WithScheme(scheme).WithObjects(
+				fakeCASecret("some-ca-secret", "some-namespace", "some-new-data"),
+			).Build(),
+			config:       fakeCAInjectionRuntimeExtensionConfig("some-extension-config", "some-namespace", "some-namespace/some-ca-secret", "some-old-ca-data"),
+			wantCABundle: []byte(`some-new-data`),
+			wantErr:      false,
+		},
+		{
+			name:    "Fail because secret does not exist",
+			client:  fake.NewClientBuilder().WithScheme(scheme).WithObjects().Build(),
+			config:  fakeCAInjectionRuntimeExtensionConfig("some-extension-config", "some-namespace", "some-namespace/some-ca-secret", ""),
+			wantErr: true,
+		},
+		{
+			name: "Fail because secret does not contain a ca.crt",
+			client: fake.NewClientBuilder().WithScheme(scheme).WithObjects(
+				fakeCASecret("some-ca-secret", "some-namespace", ""),
+			).Build(),
+			config:  fakeCAInjectionRuntimeExtensionConfig("some-extension-config", "some-namespace", "some-namespace/some-ca-secret", ""),
+			wantErr: true,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if err := reconcileCABundle(context.TODO(), tt.client, tt.config); (err != nil) != tt.wantErr {
+				t.Errorf("reconcileCABundle() error = %v, wantErr %v", err, tt.wantErr)
+			}
+			g := NewWithT(t)
+
+			err := reconcileCABundle(context.TODO(), tt.client, tt.config)
```

Comment:
```suggestion
			err := reconcileCABundle(context.Background(), tt.client, tt.config)
```
context.TODO is meant to be used in places where we eventually want to use a real context. In tests we just don't care